### PR TITLE
dbr quant overhead[7/x]: speed up AutoQuantizationState.reset_to_new_call

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68432 dbr quant overhead[9/x]: precalculate when to skip op_convert_after_hook
* #68431 dbr quant: remove more asserts from hot paths
* #68404 speed up quantized relu6 inplace kernel
* #68374 dbr quant: simplify relatedness logic
* #68373 dbr quant overhead[8/x]: small speedup in op_needs_quantization
* **#68372 dbr quant overhead[7/x]: speed up AutoQuantizationState.reset_to_new_call**
* #68371 dbr quant overhead[6/x]: remove unneeded isinstance checks in `op_convert_before_hook`
* #68370 dbr quant overhead[5/x]: remove unnecessary asserts
* #68369 dbr quant: clean up key types in AutoQuantizationState mappings
* #68351 dbr quant overhead[4/x]: speed up hook type calculations
* #68350 dbr quant overhead[3/x]: speed up AutoQuantizationState.mark_cur_op_complete

Summary:

Speeds up `AutoQuantizationState.reset_to_new_call` by going around
the getattr and setattr overhead in `torch.nn.Module`.

Test plan:

```
// MobileNetV2, 1x3x224x224 input, % of time spent by function during DBR convert

// before
reset_to_new_call - 1.09%

// after
reset_to_new_call - 0.18%
```

Differential Revision: [D32463759](https://our.internmc.facebook.com/intern/diff/D32463759)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463759).

File: torch/ao/quantization/_dbr/quantization_state.py [hunk 1]
```diff
@@ -157,7 +157,9 @@ def reset_to_new_call(self):
         """
         Resets the internal op counter to start a new top level module call
         """
-        self.idx = 0
+        # torch.nn.Module __setattr__ has overhead,
+        # this code is the explicit fast path for `self.idx = 0`
+        object.__setattr__(self, 'idx', 0)
```
Comment (jerryzh168) [inline torch/ao/quantization/_dbr/quantization_state.py:162]: nit: same comment as before, maybe we can have a set of utility functions for these things

**Review by jerryzh168 — APPROVED** [review]

File: torch/ao/quantization/_dbr/quantization_state.py (hunk 1 above)
Comment (vkuzo) [reply torch/ao/quantization/_dbr/quantization_state.py:162]: added a comment on the reasoning, not sure if adding one more function would increase the overhead, hope that's ok!

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463759).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463759).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463759).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463759).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32463759).

**Pull request closed without merging**